### ISSUE TITLE
peribolos job fails silently, presubmit job does not appear to validate properly.

### ISSUE THREAD

**Issue by vaikas** [knative/test-infra#2652]
We tried to remove a group as per here:
https://github.com/knative/community/issues/91

And the PR is here:
https://github.com/knative/community/pull/425

Notice the presubmit job ran successfully here:
https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative_community/425/pull-knative-peribolos/1356294227448827904

But looking at the logs, it doesn't really seem to do anything useful since it skips everything?

```
{"component":"unset","file":"prow/flagutil/github.go:75","func":"k8s.io/test-infra/prow/flagutil.(*GitHubOptions).Validate","level":"warning","msg":"It doesn't look like you are using ghproxy to cache API calls to GitHub! This has become a required component of Prow and other components will soon be allowed to add features that may rapidly consume API ratelimit without caching. Starting May 1, 2020 use Prow components without ghproxy at your own risk! https://github.com/kubernetes/test-infra/tree/master/ghproxy#ghproxy","severity":"warning","time":"2021-02-01T17:32:02Z"}
{"client":"github","component":"unset","file":"prow/github/client.go:594","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"Throttle(300, 100)","severity":"info","time":"2021-02-01T17:32:02Z"}
{"component":"unset","file":"prow/cmd/peribolos/main.go:819","func":"main.configureOrg","level":"info","msg":"Skipping org metadata configuration","severity":"info","time":"2021-02-01T17:32:02Z"}
{"component":"unset","file":"prow/cmd/peribolos/main.go:831","func":"main.configureOrg","level":"info","msg":"Skipping org member configuration","severity":"info","time":"2021-02-01T17:32:02Z"}
{"component":"unset","file":"prow/cmd/peribolos/main.go:838","func":"main.configureOrg","level":"info","msg":"Skipping org repositories configuration","severity":"info","time":"2021-02-01T17:32:02Z"}
{"component":"unset","file":"prow/cmd/peribolos/main.go:844","func":"main.configureOrg","level":"info","msg":"Skipping team and team member configuration","severity":"info","time":"2021-02-01T17:32:02Z"}
{"component":"unset","file":"prow/cmd/peribolos/main.go:205","func":"main.main","level":"info","msg":"Finished syncing configuration.","severity":"info","time":"2021-02-01T17:32:02Z"}
``` 

Afterwards, in a followup PR here:
https://github.com/knative/community/pull/430

Presubmit passes again with flying colors (though, by this point the configuration is in a bad state as per above):
https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative_community/430/pull-knative-peribolos/1357246540430708736

```
{"component":"unset","file":"prow/flagutil/github.go:75","func":"k8s.io/test-infra/prow/flagutil.(*GitHubOptions).Validate","level":"warning","msg":"It doesn't look like you are using ghproxy to cache API calls to GitHub! This has become a required component of Prow and other components will soon be allowed to add features that may rapidly consume API ratelimit without caching. Starting May 1, 2020 use Prow components without ghproxy at your own risk! https://github.com/kubernetes/test-infra/tree/master/ghproxy#ghproxy","severity":"warning","time":"2021-02-04T08:36:15Z"}
{"client":"github","component":"unset","file":"prow/github/client.go:594","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"Throttle(300, 100)","severity":"info","time":"2021-02-04T08:36:15Z"}
{"component":"unset","file":"prow/cmd/peribolos/main.go:819","func":"main.configureOrg","level":"info","msg":"Skipping org metadata configuration","severity":"info","time":"2021-02-04T08:36:15Z"}
{"component":"unset","file":"prow/cmd/peribolos/main.go:831","func":"main.configureOrg","level":"info","msg":"Skipping org member configuration","severity":"info","time":"2021-02-04T08:36:15Z"}
{"component":"unset","file":"prow/cmd/peribolos/main.go:838","func":"main.configureOrg","level":"info","msg":"Skipping org repositories configuration","severity":"info","time":"2021-02-04T08:36:15Z"}
{"component":"unset","file":"prow/cmd/peribolos/main.go:844","func":"main.configureOrg","level":"info","msg":"Skipping team and team member configuration","severity":"info","time":"2021-02-04T08:36:15Z"}
{"component":"unset","file":"prow/cmd/peribolos/main.go:205","func":"main.main","level":"info","msg":"Finished syncing configuration.","severity":"info","time":"2021-02-04T08:36:15Z"}
```

And any subsequent PR is now in a wonky state, for example:
https://prow.knative.dev/view/gs/knative-prow/logs/post-knative-peribolos/1357247051569565696

Things are now failing silently:

```
{"component":"unset","file":"prow/flagutil/github.go:75","func":"k8s.io/test-infra/prow/flagutil.(*GitHubOptions).Validate","level":"warning","msg":"It doesn't look like you are using ghproxy to cache API calls to GitHub! This has become a required component of Prow and other components will soon be allowed to add features that may rapidly consume API ratelimit without caching. Starting May 1, 2020 use Prow components without ghproxy at your own risk! https://github.com/kubernetes/test-infra/tree/master/ghproxy#ghproxy","severity":"warning","time":"2021-02-04T08:38:13Z"}
{"client":"github","component":"unset","file":"prow/github/client.go:594","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"Throttle(1200, 100)","severity":"info","time":"2021-02-04T08:38:13Z"}
{"client":"github","component":"unset","file":"prow/github/client.go:594","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"GetOrg(knative)","severity":"info","time":"2021-02-04T08:38:13Z"}
{"client":"github","component":"unset","file":"prow/github/client.go:594","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"ListOrgInvitations(knative)","severity":"info","time":"2021-02-04T08:38:13Z"}
{"client":"github","component":"unset","file":"prow/github/client.go:594","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"User()","severity":"info","time":"2021-02-04T08:38:13Z"}
{"client":"github","component":"unset","file":"prow/github/client.go:594","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"ListOrgMembers(knative, admin)","severity":"info","time":"2021-02-04T08:38:13Z"}
{"client":"github","component":"unset","file":"prow/github/client.go:594","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"ListOrgMembers(knative, member)","severity":"info","time":"2021-02-04T08:38:13Z"}
{"component":"unset","file":"prow/cmd/peribolos/main.go:202","func":"main.main","level":"fatal","msg":"Configuration failed: failed to configure knative members: all team members/maintainers must also be org members: eallred-google","severity":"fatal","time":"2021-02-04T08:38:16Z"}
```

Net effect is that it looks like:
* Pre-submit does not validate properly
* This leads to hilarity once a bad config goes in, because now any subsequent PR will also fail
* The failures are invisible because subsequent PRs don't validate and while applying the new configuration failures are invisible so it's impossible to know what state the system is without actually going through the prow logs.

**Comment by vaikas** [comment]
Ok, so after moving @eallred-google to org member, it appears that there was a larger number of actions taken by the postjob, judging by the logs here:
https://prow.knative.dev/view/gs/knative-prow/logs/post-knative-peribolos/1357270470004576256

Some of the items in the logs seem a bit unsettling however:
```
{"client":"github","component":"unset","file":"prow/github/client.go:594","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"RemoveTeamRepo(3996822, knative, client)","severity":"info","time":"2021-02-04T10:18:26Z"}
{"client":"github","component":"unset","file":"prow/github/client.go:594","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"RemoveTeamRepo(3996822, knative, test-infra)","severity":"info","time":"2021-02-04T10:18:29Z"}
{"client":"github","component":"unset","file":"prow/github/client.go:594","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"RemoveTeamRepo(3646480, knative, networking)","severity":"info","time":"2021-02-04T10:22:32Z"}
{"client":"github","component":"unset","file":"prow/github/client.go:594","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"RemoveTeamRepo(3646480, knative, serving)","severity":"info","time":"2021-02-04T10:22:35Z"}
{"client":"github","component":"unset","file":"prow/github/client.go:594","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"RemoveTeamRepo(3646480, knative, pkg)","severity":"info","time":"2021-02-04T10:22:38Z"}

```
Unfortunately there's no history to see when the last successful run of this was:
https://prow.knative.dev/?job=*post-knative-peribolos*

I will raise visibility of this just as a heads up in case badness actually resulted from any of these actions.

**Comment by vaikas** [comment]
Also tried to look a the periodic run logs, since it runs every day at 1700, but I can only see two runs there.
https://prow.knative.dev/?job=ci-knative-peribolos

Is there any other place to look for logs to see how long things have been b0rk3d?

**Comment by vaikas** [comment]
Ok, went to gcs to grab some older logs. This for example:
gs://knative-prow/logs/post-knative-peribolos/1340847225189699584/build-log.txt

Shows all those kinds of RemoveTeamRepo messages, so just looking briefly it does look like it always prints those (phew):

```
{"client":"github","component":"unset","file":"prow/github/client.go:594","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"RemoveTeamRepo(3646485, knative, serving)","severity":"info","time":"2020-12-21T02:43:25Z"}
{"client":"github","component":"unset","file":"prow/github/client.go:594","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"RemoveTeamRepo(3646485, knative, pkg)","severity":"info","time":"2020-12-21T02:43:28Z"}
{"client":"github","component":"unset","file":"prow/github/client.go:594","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"RemoveTeamRepo(3646485, knative, caching)","severity":"info","time":"2020-12-21T02:43:31Z"}
{"client":"github","component":"unset","file":"prow/github/client.go:594","func":"k8s.io/test-infra/prow/github.(*client).log","level":"info","msg":"RemoveTeamRepo(3646485, knative, networking)","severity":"info","time":"2020-12-21T02:43:34Z"}
```
Problem of silent failures is still obviously a problem, but after unblocking the runs by fixing the eallred-google entry, things seem to be just like before.

**Comment by vaikas** [comment]
Just an update on this. Looks like the last PR we did catch a similar issue that previously (the original pointer in this issue) was not caught:
https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative_community/455/pull-knative-peribolos/1361402784523489280

```
{"component":"unset","file":"prow/cmd/peribolos/main.go:202","func":"main.main","level":"fatal","msg":"Configuration failed: failed to configure knative members: users in both roles: rhuss","severity":"fatal","time":"2021-02-15T19:51:44Z"}
```

So, I'm going to close this issue since we have an #2662 tracking post-submit failure notifications.
